src/librustc_metadata/loader.rs

```diff
@@ -302,14 +302,11 @@ impl<'a> Context<'a> {
     pub fn load_library_crate(&mut self) -> Library {
         match self.find_library_crate() {
```

Comment:
Nit: This could become `self.find_library_crate().unwrap_or_else(|| self.report_load_errs())`.